sql/core/src/main/scala/org/apache/spark/sql/execution/python/AttachDistributedSequenceExec.scala

```diff
@@ -63,6 +97,16 @@ case class AttachDistributedSequenceExec(
     }
   }
 
+  override protected[sql] def cleanupResources(): Unit = {
+    this.synchronized {
+      if (cached != null && cached.getStorageLevel != StorageLevel.NONE) {
```

Comment:
are all the accesses of `cached` protected by `synchronized`?

Reply:
I think so, maybe we don't need the `synchronized `

Reply:
Yea, `cached` starts with `null`, and get assigned during `doExecute`. Seems `@transient` is sufficient.

Reply:
let me update